README.md
# expresso

<img alt="Coffee Cup Icon by Freepik on Flaticon" src="coffee-cup.svg" style="float: center;" width=200 height=200 />

[![GitHub issues](https://img.shields.io/github/issues/Shrubhog/expresso.svg)](https://github.com/Shrubhog/expresso/issues) 
[![GitHub forks](https://img.shields.io/github/forks/Shrubhog/expresso.svg)](https://github.com/Shrubhog/expresso/network)
[![GitHub stars](https://img.shields.io/github/stars/Shrubhog/expresso.svg)](https://github.com/Shrubhog/expresso/stargazers)
[![GitHub license](https://img.shields.io/github/license/Shrubhog/expresso.svg)](https://github.com/Shrubhog/expresso/blob/master/LICENSE) [![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg?style=flat-square)](http://makeapullrequest.com)


Incremental game built entirely around coffee. It's pretty small. I'd really like some help on more features, so pull requests are very much appreciated.

## What I've done
Technology tree goes like this:
* Scrounge for money in the couch, button disables itself after 10 dollars, gives one dollar
* Do a favor, no more money in the couch, appears after 10 dollars, disables after 100, gives 10 dollars
* Get a coffee machine for 100 dollars, gives a coffee machine
* Make coffee for neighbors, gives 5 dollars
* Automate coffee machine for 50 dollars, gives one auto coffee machine (gives $5/second), costs one coffee machine
* Grandma's basement, because you need somewhere to set up shop, and your parents said no. $200
* Interns, which help around and just do stuff. Costs $50. $10/s
* Croissant Makers give you $20/10s. Cost $75.
* Cookie Makers give you $35/10s. Cost $95.
## Plans
I'm planning to switch from plain HTML, JS, and CSS to Angular and Typescript.
The main reason is because I've seen a lot of incremental games using Angular routes, which I think are pretty amazing.
But right now, I'm gonna stay in HTML, JS, and CSS because I want to add some more features:
* Upgrades: coffee flavours, croissant flavours
* Donut Makers
* Shops increase overall income by 5%

## Changelog
There is a changelog [here](CHANGELOG.md). It is a compiled list of updates since `v.0.1`.

> &copy; Coffee Cup Icon by [Freepik](freepik.com) on [Flaticon](flaticon.com)


CHANGELOG.md
## Changelog
A list of versions since v.0.1.

Based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0), and *kind of* adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]
- Tabs: Info, Shop, Upgrades
## [0.2] - 2018-11-16
### Added
- Dollars per Second in the info
- Added Favicon
### Canceled
- Percentage DpS in tooltips

## [0.1] - 2018-11-13
### Added
- Changelog file
- Basically nothing. Just a pointer to the 52nd commit of expresso.

main.js
// expresso: main.js
// v. 0.1
// by Shrubhog

// Money, the stablest and most unchanging currency in the world. 
var money = 0;

// Crude updating tooltip, because ${} requires you to reload the variable after every price. Again, and again, and again. 
var tooltips = {
  coffee: "Costs $-, allows you to make coffee. It's pretty good coffee.",
  autoCoffee: "Costs $- and a coffee machine, gives an auto coffee machine and $1/s. Making coffee is pretty slow. Let's make the robots do it!",
  intern: "Costs $-. Gets you an intern to help around and make coffee. $10/s",
  croissants: "Costs $-. Hire a person to make those little french bagel things. $20/10s",
  cookies: "Costs $-. Hire a nice baker to make the odd batch of cookies, and believe me, they are AMAZING cookies. $35/10s"
}

// another crude solution to remember how many times the buttons are pressed.
var couchSearched = 0;
var favorsDone = 0;

// functions

function getPaid(num) {
  money += num;
  $("#money").html(money);
}
function couch() {
  if (couchSearched < 9) {
    getPaid(1);
    couchSearched += 1;
  } else if (couchSearched == 9) {
    getPaid(1);
    couchSearched += 1;
    $("#couch").addClass("disabledButton");
    $( "#couch" ).tooltip( "option", "disabled", true );
  } else if (couchSearched >= 10) {
    $("#couch").addClass("disabledButton");
  }
}
function favor() {
  if (favorsDone < 8) {
    getPaid(10);
    favorsDone += 1;
  } else if (favorsDone == 8) {
    getPaid(10);
    favorsDone += 1;
    $("#favor").addClass("disabledButton");
    $( "#favor" ).tooltip( "option", "disabled", true );
  } else if (couchSearched >= 9) {
    $("#couch").addClass("disabledButton");
  }
}
var coffeeMachines = 0;
function buyCoffeeMachines(){

    var coffeeMachineCost = Math.floor(100 * Math.pow(1.1,coffeeMachines));
    if(money >= coffeeMachineCost){
        if (typeof $("#coffeeMachines").html() == "undefined") {
          $(".info").append("Coffee Machines: <span id='coffeeMachines'></span><br>")
          $("#coffeeMachines").html(coffeeMachines)
        }
        coffeeMachines = coffeeMachines + 1;
    	  money = money - coffeeMachineCost;
        $('#coffeeMachines').html(coffeeMachines);
        $('#money').html(money);
        var nextCost = Math.floor(100 * Math.pow(1.1,coffeeMachines));
        var priceTooltip1 = tooltips.coffee.split("-");
        var priceTooltip2 = priceTooltip1[0] + nextCost + priceTooltip1[1];
        $("button#buyCoffeeMachine").attr("title", priceTooltip2)
    };
    
};

var autoCoffees = 0;
function buyAutoCoffee() {
    var autoCoffeeCost = Math.floor(50 * Math.pow(1.1,autoCoffees));
    console.log(autoCoffeeCost)
    if(money >= autoCoffeeCost && coffeeMachines > 0){
        console.log("Can buy it!")
        if (typeof $("#autoCoffees").html() == "undefined") {
          $(".info").append("Auto Coffee Machines: <span id='autoCoffees'></span><br>")
          $("#autoCoffees").html(autoCoffees)
        }
        autoCoffees += 1;
    	  money = money - autoCoffeeCost;
        coffeeMachines = coffeeMachines - 1;
        $('#autoCoffees').html(autoCoffees);
        $('#coffeeMachines').html(coffeeMachines);
        $('#money').html(money);
        var nextCost = Math.floor(50 * Math.pow(1.1,autoCoffees));
        var priceTooltip1 = tooltips.autoCoffee.split("-");
        var priceTooltip2 = priceTooltip1[0] + nextCost + priceTooltip1[1];
        $("button#buyAutoCoffee").attr("title", priceTooltip2)
    };
    
};
var interns = 0;
function hireIntern() {
  if (haveBasement) {
    $("#hireIntern").removeClass("disabledButton");
    $("#hireIntern").addClass("Button");
    var internCost = Math.floor(50 * Math.pow(1.1,interns));
    console.log(internCost)
    if(money >= internCost){
        console.log("Can buy it!")
        if (typeof $("#interns").html() == "undefined") {
          $(".info").append("Interns: <span id='interns'></span><br>")
          $("#interns").html(interns)
        }
        interns += 1;
    	  money = money - internCost;
        $('#interns').html(interns);
        $('#money').html(money);
        var nextCost = Math.floor(50 * Math.pow(1.1,interns));
        var priceTooltip1 = tooltips.intern.split("-");
        var priceTooltip2 = priceTooltip1[0] + nextCost + priceTooltip1[1];
        $("button#hireIntern").attr("title", priceTooltip2)
    };
  }
};

var croissants = 0;

function hireCroissantMaker() {
  if (haveBasement) {
    $("#hireCroissantMaker").removeClass("disabledButton");
    $("#hireCroissantMaker").addClass("Button");
    var croissantCost = Math.floor(75 * Math.pow(1.1,croissants));
    console.log(croissantCost)
    if(money >= croissantCost){
        console.log("Can buy it!")
        if (typeof $("#croissant").html() == "undefined") {
          $(".info").append("Croissant Makers: <span id='croissant'></span><br>")
          $("#croissant").html(interns)
        }
        croissants += 1;
    	  money = money - croissantCost;
        $('#croissant').html(croissants);
        $('#money').html(money);
        var nextCost = Math.floor(75 * Math.pow(1.1,croissants));
        var priceTooltip1 = tooltips.croissants.split("-");
        var priceTooltip2 = priceTooltip1[0] + nextCost + priceTooltip1[1];
        $("button#hireCroissantMaker").attr("title", priceTooltip2)
    };
  }
};

var cookies = 0;
function hireCookieMaker() {
  if (haveBasement) {
    $("#hireCookieMaker").removeClass("disabledButton");
    $("#hireCookieMaker").addClass("Button");
    var cookieCost = Math.floor(95 * Math.pow(1.1,cookies));
    console.log(cookieCost)
    if(money >= cookieCost){
        console.log("Can buy it!")
        if (typeof $("#cookie").html() == "undefined") {
          $(".info").append("Cookie Makers: <span id='cookie'></span><br>")
          $("#cookie").html(cookies)
        }
        cookies += 1;
    	  money = money - cookieCost;
        $('#cookie').html(cookies);
        $('#money').html(money);
        var nextCost = Math.floor(95 * Math.pow(1.1,cookies));
        var priceTooltip1 = tooltips.cookies.split("-");
        var priceTooltip2 = priceTooltip1[0] + nextCost + priceTooltip1[1];
        $("button#hireCookieMaker").attr("title", priceTooltip2)
    };
  }
};

var haveBasement = false;
function buyBasement() {
  if (!haveBasement && money > 200) {
    money = money - 200;
    haveBasement = true;
  }
}

// game clock, set to every quarter of a second because every second is too slow for me.

window.setInterval(function() {
  getPaid(autoCoffees * 1.375);
  getPaid(interns * 2.5);
}, 250)
// second game clock that is set to 2.5 seconds because that is four times faster than ten seconds.
window.setInterval(function() {
  getPaid(croissants * 5);
  getPaid(cookies * 8.75)
}, 2500)
// game MpS
window.setInterval(function() {
  var MpS = (autoCoffees * 5.5)+(interns * 2.5)+(croissants * 2)+(cookies * 3.5);
  $("#MpS").html(MpS);
}, 250)

$(document).on("keypress", function (e) {
  if (String.fromCharCode(e.which) == "i") {
    $("#tab-info").tab("show");
  } else if (String.fromCharCode(e.which) == "s") {
    $("#tab-shop").tab("show");
  } else if (String.fromCharCode(e.which) == "u") {
    $("#tab-upgrades").tab("show");
  }
});